### PR TITLE
Add Marshal and Unmarshal functions to facilitate converting to bytes

### PR DESCRIPTION
**Pull request by aignas**
Most likely the user of this library will have a request/response data structure, which they will want to pass around as bytes. With this change we make it easier to do so by handling the creation of `*bitio.Count{Reader,Writer}` and decrease the likelihood that the user forgets to close the writer and gets incomplete data. This also allows us to introduce a new function, which accepts an interface and can be easier to ensure that it will work between different versions of the libraries and the generated code.

Note that a similar method is present in the Python implementation.

Summary:
- [X] Add new functions for ease of use
- [X] Migrate integration tests to the new functions
- [X] Migrate the example
- [X] Migrate array tests

### CODE REVIEW

**Comment by aignas** [comment]
Whilst this is not strictly a substitute for #43, it may be a useful addition to the changes that can be made in this area.

File: marshal.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+	if err := w.Close(); err != nil {
+		return nil, fmt.Errorf("flush buffer: %w", err)
+	}
+
+	return buf.Bytes(), nil
+}
+
+// Unmarshal accepts a zserio byte stream and a type and returns an error if it
+// happens during unmarshaling of the byte stream.
```
Comment (wichert) [inline marshal.go:29]: Nit: It accepts a pointer to an instance of a zserio object, not a type.

File: ztype/array_encode_test.go [hunk 2]
```diff
@@ -117,16 +113,13 @@ func TestBitfieldArrayEncoding(t *testing.T) {
...
+			got, err := zserio.Marshal(&arrayInstance)
 			if err != nil {
 				t.Fatalf("error writing array: %v", err)
 			}
 
-			if diff := cmp.Diff(test.want, buf.Bytes()); diff != "" {
+			if diff := cmp.Diff(test.want, got); diff != "" {
 				t.Errorf("incorrect encoding: %s", diff)
 			}
 		})
```
Comment (wichert) [inline ztype/array_encode_test.go:125]: There seems to be a lot of repetition here. Can we factor a function like `func verifyMarshal(t *testing.T, want []byte, obj zserio.Marshal)` ?

Comment (aignas) [reply ztype/array_encode_test.go:125]: Done.

File: marshal.go (hunk 1 above)
Comment (aignas) [reply marshal.go:29]: Done

**Review by wichert — APPROVED** [review]
Looks good!

**Pull request merged**